core/corehttp/gateway_handler.go

```diff
@@ -178,14 +178,14 @@ func (i *gatewayHandler) getOrHeadHandler(ctx context.Context, w http.ResponseWr
 		return
 	}
 
-	dr, err := i.api.Unixfs().Cat(ctx, resolvedPath)
+	dr, err := i.api.Unixfs().Get(ctx, resolvedPath)
 	dir := false
-	switch err {
-	case nil:
-		// Cat() worked
-		defer dr.Close()
-	case coreiface.ErrIsDir:
-		dir = true
+	switch {
+	case err == nil:
```

Comment:
Why change this? If we're going to go with something like this, a simple `if err == nil { ... } else { ... }` would probably be cleaner.